salt/utils/jinja.py

```diff
@@ -61,7 +61,32 @@ class SaltCacheLoader(BaseLoader):
     Templates are cached like regular salt states
     and only loaded once per loader instance.
     '''
-    def __init__(self, opts, saltenv='base', encoding='utf-8',
+
+    _singletons = {}
```

Comment:
I am pretty sure it would be garbage collected when it's reset to a new {} without need to use WeakKeyDictionary? The key is (true,) or (false,)